### ISSUE TITLE
Update header structure to meet a11y requirements

### ISSUE THREAD

**Issue by sijiayam** [civiform/civiform#9048]
Describe the issue:
- On some pages, including the review page, we are wrapping headers in <span> elements with h3 styling since we are missing h2. We need to resolve how to create accessible header structures

**Comment by sijiayam** [comment]
@ssandbekkhaug Feel free to capture any other places where we are bypassing header styles with <span> and css classes

**Comment by ssandbekkhaug** [comment]
The CSS classes that do this are:

https://github.com/civiform/civiform/blob/ca7b302b66903f1924196d536f4d8fca6093a6d8/server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss#L153

https://github.com/civiform/civiform/blob/ca7b302b66903f1924196d536f4d8fca6093a6d8/server/app/assets/stylesheets/northstar/_uswds-theme-custom-styles.scss#L162

**Comment by melanie-exygy** [comment]
I recommend that we remove those two CSS classes since they encourage engineers to break accessibility principles.  We should be following good heading hierarchy.  If the design is skipping a heading level, then we should fix the design.

@sijiayam We're using `cf-prose-h3` on the summary page and `cf-prose-h4` several places in the applicant upsell view.

**Comment by sijiayam** [comment]
@melanie-exygy agreed.

For the case of the review page we can use h3 instead of a span tag if we update the step indicator text to be an h2 as described in #9056 

Could you show me which specific applicant upsell views use the cf-prose-h4 tags?

**Comment by melanie-exygy** [comment]
@sijiayam See the yellow underlined headings in the screenshot below.  These four headings are `<h3>` that are made to appear as `<h4>`.  I did a little bit of research and I think this is actually okay, accessibility-wise.  We're using the right semantic structure, so screen reader users will be able to navigate through the headings.  It does mean that our `<h3>` heading styles may not be consistent here with other places on the website, but that doesn't really seem like a problem.
What do you think?
![Screenshot 2024-10-29 at 11 18 04 AM](https://github.com/user-attachments/assets/d11f2f8a-a739-4a1c-a67a-3d0f5204d58b)

**Comment by ssandbekkhaug** [comment]
[Program cards](https://www.figma.com/design/RmhVJCLgAKWGENJLWWAtNS/CiviForm%3A-Components%2C-Styles?node-id=4628-13624&m=dev) also have this problem. They mocks specify h4, but the application index page expects h3.

**Comment by sijiayam** [comment]
> [@sijiayam](https://github.com/sijiayam) See the yellow underlined headings in the screenshot below. These four headings are `<h3>` that are made to appear as `<h4>`. I did a little bit of research and I think this is actually okay, accessibility-wise. We're using the right semantic structure, so screen reader users will be able to navigate through the headings. It does mean that our `<h3>` heading styles may not be consistent here with other places on the website, but that doesn't really seem like a problem. What do you think? ![Screenshot 2024-10-29 at 11 18 04 AM](https://github.com/user-attachments/assets/d11f2f8a-a739-4a1c-a67a-3d0f5204d58b)

I am okay with this. Alright for visual styles to differ too support screen reader navigation.

Since there is no longer an issue here I will close the issue.
